### PR TITLE
Deprecate python 3.7

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Since 3.7 EoL is two weeks away, this PR:

- Modifies the README to say we support 3.8+
- Removes 3.7 from pyproject.toml files.
- Import `Literal` from `typing` as opposed to `typing_extensions` since it was introduced in 3.8
- Removes 3.7 from CI. Rather than having 3.8 and 3.9 tests, just running tests in 3.8 so that we can cut down the number of CI jobs by two.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Gradio is useful for:
 
 ## Quickstart
 
-**Prerequisite**: Gradio requires Python 3.7 or higher, that's all!
+**Prerequisite**: Gradio requires Python 3.8 or higher, that's all!
```
Comment (abidlabs) [inline README.md:42]: There's a few other places where we mention 3.7:
- README badge (both on the main README as well as on the Chinese version)
- Contributing.md

Can we knock those out too?

Comment (abidlabs) [reply README.md:42]: We also mention in `01_quickstart.md`

File: test/requirements-37.txt
Comment (abidlabs) [inline test/requirements-37.txt:1]: We'll need to regenerated `test/requirements.txt` for Python 3.8 (right now, it's generated for 3.9)

File: gradio/blocks.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import webbrowser
 from abc import abstractmethod
 from types import ModuleType
-from typing import TYPE_CHECKING, Any, AsyncIterator, Callable
+from typing import TYPE_CHECKING, Any, AsyncIterator, Callable, Literal
```
Comment (abidlabs) [inline gradio/blocks.py:15]: We also use `typing_extensions` a few times in the python client. Would be good to get rid of those too, as well as the dependency`typing_extensions` altogether from the client

**Comment by abidlabs** [comment]
Thanks @freddyaboulton! Spotted a few more opportunities to wipe out any 3.7 dependencies. Should be good to go once we remove those

File: gradio/blocks.py (hunk 2 above)
Comment (freddyaboulton) [reply gradio/blocks.py:15]: We need typing_extensions for `NotRequired` still!

File: test/requirements-37.txt
Comment (freddyaboulton) [reply test/requirements-37.txt:1]: The tests are passing though so I don't think we need to.

File: README.md (hunk 1 above)
Comment (freddyaboulton) [reply README.md:42]: Good catch - pushed this up now!

File: gradio/blocks.py (hunk 2 above)
Comment (abidlabs) [reply gradio/blocks.py:15]: Gotcha. Looking forward to Gradio being Python 3.11+!

**Comment by freddyaboulton** [comment]
Thanks @abidlabs ! The client already works on 3.8 so I just edited the README to mention that it supports 3.8.

**Comment by freddyaboulton** [comment]
That's already done in this PR @abidlabs !

**Comment by abidlabs** [comment]
Don't we need to update line 137 in .github/workflows/backend.yml?

**Comment by abidlabs** [comment]
Wait I might be confused

**Comment by abidlabs** [comment]
Oh that's just the linter. But I guess we should update that too. 

Otherwise looks good to merge!

**Comment by freddyaboulton** [comment]
Yea that's just the lint! Just pushed up to be 3.8

**Comment by freddyaboulton** [comment]
Thanks for the review @abidlabs !

**Pull request merged**